controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -590,21 +256,9 @@ func (r *StorageConsumerReconciler) verifyCephResourcesDoNotExist() bool {
 			cephClient := &rookCephv1.CephClient{}
 			cephClient.Name = cephResource.Name
 			cephClient.Namespace = r.namespace
-			if err := r.get(cephClient); err == nil || !errors.IsNotFound(err) {
-				return false
-			}
-		case "CephBlockPool":
-			cephBlockPool := &rookCephv1.CephBlockPool{}
-			cephBlockPool.Name = cephResource.Name
-			cephBlockPool.Namespace = r.namespace
-			if err := r.get(cephBlockPool); err == nil || !errors.IsNotFound(err) {
-				return false
-			}
-		case "CephFilesystemSubVolumeGroup":
-			cephFilesystemSubVolumeGroup := &rookCephv1.CephFilesystemSubVolumeGroup{}
-			cephFilesystemSubVolumeGroup.Name = cephResource.Name
-			cephFilesystemSubVolumeGroup.Namespace = r.namespace
-			if err := r.get(cephFilesystemSubVolumeGroup); err == nil || !errors.IsNotFound(err) {
+			err := r.get(cephClient)
+			cephUserType := cephClient.Annotations[StorageCephUserTypeAnnotation]
+			if cephUserType == "healthchecker" && (err == nil || !errors.IsNotFound(err)) {
```

Comment:
Why are you expecting other cephUserTypes here? if the watch is an Own watch, only the healthchecker clients will trigger this code. 

Please revert to the prev code.

Reply:
ack